flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java

```diff
@@ -188,9 +189,12 @@ public WebRuntimeMonitor(
 
 			// Cancel a job via GET (for proper integration with YARN this has to be performed via GET)
 			.GET("/jobs/:jobid/yarn-cancel", handler(new JobCancellationHandler()))
-			// DELETE is the preferred way of cancelling a job (Rest-conform)
+			// DELETE is the preferred way of canceling a job (Rest-conform)
 			.DELETE("/jobs/:jobid", handler(new JobCancellationHandler()))
 
+			// stop a job
+			.DELETE("/jobs/:jobid/stop", handler(new JobStoppingHandler()))
+
```

Comment:
This is somewhat counter-intuitive. `DELETE` is meant to remove the resource at the specified URI. Specifying a action as `/stop` is not a good idea. Perhaps a better idea would be to identify these actions as `/jobs/:jobid?mode=cancel` and `/jobs/:jobid?mode=stop`. @StephanEwen might have a better idea about this.